spec_util/meld.go

```diff
@@ -510,7 +580,7 @@ func (m *melder) meldList(dst, src *pb.List) error {
 // Assumes dst.value == src.value.
 // Meld data formats, tracking data, etc. from src to dst.
 // XXX(cns): In some cases, this modifies src as well as dst :/
-func (m *melder) meldPrimitive(dst, src *pb.Primitive) {
+func (m *melder) meldPrimitive(dst, src *pb.Primitive) error {
```

Comment:
When can we expect this to (not) return an error? When `haveCompatibleTypes(dstPrim, srcPrim)`? Comment should say. (Or better—but less scrappy—roll `haveCompatibleTypes` into this method somehow.)

Reply:
I considered it but didn't want to do more refactoring.  `meldPrimitive` is called exactly once and isn't exposed outside this package, so I'm not too worried.  I updated the comment.